### ISSUE TITLE
istio convert grpc to http1.1 when server side inbound injection is disabled

### ISSUE THREAD

**Issue by coreyzzp** [istio/istio#27271]
**Bug description**

This is the test enviorment 

```
Pod A (ghz / enovy) ->  Pod B (enovy/Server)
```

I want to test the performance of both the server and istio proxy, and the CPU shows  both the server and ghz reach about
100% CPU usage while enovy takes up to 200% CPU for both side of the test, so I try to disable inbound injection ( because i think it a way to reduce CPU overhead introduced by enovy, since the request has reached destination pod, there is no need to go through almost the same packet processing path as the outbound enovy in PodA, is that right? )of the 
Server side enovy  by config followings annotions  (In Pod B 's deployment yaml), 

```
traffic.sidecar.istio.io/includeInboundPorts: ""
```

then when I launch test again from Pod A, the tool reports that 

```
  [1]   rpc error: code = Unavailable desc = upstream connect error or disconnect/reset before headers. reset reason: connection termination
```

when I try to tcpdump the packet from the case, I found my grpc packet become Http1.1 Post request !? 

when ghz is running in Pod without enovy, the test works fine again.

what it looks like 
![image](https://user-images.githubusercontent.com/43430714/92992740-b6ed0780-f51f-11ea-9a82-3c5d2cafad01.png)

before
![image](https://user-images.githubusercontent.com/43430714/92992736-aa68af00-f51f-11ea-958e-40aaeb4d8743.png)

where 50051 is the grpc target port and the port is configed as a grpc prefix in service yaml

```
apiVersion: v1
kind: Service
spec:
  ports:
  - name: grpc-my  
    port: 50051
    protocol: TCP
    targetPort: grpc
```

[ ] Docs
[ ] Installation
[X] Networking
[ ] Performance and Scalability
[ ] Extensions and Telemetry
[ ] Security
[ ] Test and Release
[X] User Experience
[ ] Developer Infrastructure

**Expected behavior**

when server side injection is disabled, grpc should works as before

**Steps to reproduce the bug**

1. disable inbound grpc injection
2. request grpc from another pod

**Version (include the output of `istioctl version --remote` and `kubectl version --short` and `helm version` if you used Helm)**

Client Version: v1.14.3-tke.14
Server Version: v1.14.3-tke.14
Istio 1.5.6

**How was Istio installed?**

helm

**Environment where bug was observed (cloud vendor, OS, etc)**

**Comment by howardjohn** [comment]
Fixed by https://github.com/istio/istio/pull/24712. Please upgrade to 1.7+. Thanks!

**Comment by coreyzzp** [comment]
> Fixed by #24712. Please upgrade to 1.7+. Thanks!

the problem is solved in that version by explicitly setting the istio/dr tls disabled, it seems to be enabled by default
